packages/react-instantsearch-hooks-dom/src/widgets/__tests__/SortBy.test.tsx

```diff
@@ -0,0 +1,192 @@
+    );
+
+    await wait(0);
+
+    expect(document.querySelector('.ais-SortBy-select')).toHaveValue(
+      'instant_search'
+    );
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="ais-SortBy"
+        >
+          <select
+            class="ais-SortBy-select"
+          >
+            <option
+              class="ais-SortBy-option"
+              value="instant_search"
+            >
+              Featured
+            </option>
+            <option
+              class="ais-SortBy-option"
+              value="instant_search_price_asc"
+            >
+              Price (asc)
+            </option>
+            <option
+              class="ais-SortBy-option"
+              value="instant_search_price_desc"
+            >
+              Price (desc)
+            </option>
+          </select>
+        </div>
+      </div>
+    `);
+  });
+
+  test('transform the passed items', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper>
+        <SortBy
+          transformItems={(items) =>
+            items.map((item) => ({
+              ...item,
+              label: item.label.toUpperCase(),
+            }))
+          }
+          items={[
+            { label: 'Featured', value: 'instant_search' },
+            { label: 'Price (asc)', value: 'instant_search_price_asc' },
+            { label: 'Price (desc)', value: 'instant_search_price_desc' },
+          ]}
+        />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await wait(0);
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="ais-SortBy"
+        >
+          <select
+            class="ais-SortBy-select"
+          >
+            <option
+              class="ais-SortBy-option"
+              value="instant_search"
+            >
+              FEATURED
+            </option>
+            <option
+              class="ais-SortBy-option"
+              value="instant_search_price_asc"
+            >
+              PRICE (ASC)
+            </option>
+            <option
+              class="ais-SortBy-option"
+              value="instant_search_price_desc"
+            >
+              PRICE (DESC)
+            </option>
+          </select>
+        </div>
+      </div>
+    `);
+  });
+
+  test('updates the selected index', async () => {
+    const { getByRole } = render(
+      <InstantSearchHooksTestWrapper>
+        <SortBy
+          items={[
+            { label: 'Featured', value: 'instant_search' },
+            { label: 'Price (asc)', value: 'instant_search_price_asc' },
+            { label: 'Price (desc)', value: 'instant_search_price_desc' },
+          ]}
+        />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await wait(0);
+
+    userEvent.selectOptions(
+      document.querySelector('.ais-SortBy-select') as HTMLSelectElement,
+      getByRole('option', { name: 'Price (asc)' })
+    );
+
+    await wait(0);
```

Comment:
we want to see that the search results also come from that other index, otherwise you could have a select without refine still passing this test

Reply:
Addressed in 810473f3.